### DISCUSSION TITLE
Tiff to PNG darker compared to Photoshop

### DISCUSSION THREAD

**Discussion by le0daniel** [ImageMagick/ImageMagick#3707] [Help]
Hi all. First of all, thx for any help.

I m using Imagemagick (7.0.10-57) on Linux to convert a tiff ([color_small.tif.zip](https://github.com/ImageMagick/ImageMagick/files/6510372/color_small.tif.zip)) to a smaller PNG file. I use following command:
`convert -resize 2400x\> file.tif[0] -colorspace sRGB PNG:targetfile.png` 
I tried lot's of different options (adding profiles, PNG64, quality, flatten) but still, the image is darker than what photoshop exports.

I read through a lot of old forum posts, but did not find any answer to why this is the case and how to resolve it. 
The outputted files ([Archive.zip](https://github.com/ImageMagick/ImageMagick/files/6510400/Archive.zip)). If you compare them side by side, the image is visually darker. I am a bit lost and out of options. I have no clue what the issue is or what photoshop does differently. The Photoshop file is slightly bigger and has an alpha channel. Adding the options to force imagemagick to a PNG64 does not change anything.

ImageMagick: 
![out](https://user-images.githubusercontent.com/11374766/118850591-3c0b9600-b8d1-11eb-9f88-d4cc73a7555c.png)

Photoshop:
![photoshop 2](https://user-images.githubusercontent.com/11374766/118850620-4332a400-b8d1-11eb-915c-ac57660fa899.png)

**Comment by fmw42** [comment]
I loaded your tiff file in Photoshop CS then saved as PNG. I then converted your tiff file with IM 6.9.11.13 Q16 Mac OSX Sierra and LIBTIFF, Version 4.1.0 as follows and did a compare. They are exactly the same.

(Note that you should read the input before doing -resize for proper syntax, though IM 6 is forgiving. But IM 7 is not)

```
convert -quiet color_small.tif color_small_im.png

compare -metric rmse color_small_im.png color_small_ps.png null:
0 (0)
```

Check your version of libtiff to be sure it is current.

Also the resize will likely give slightly different results due to different interpolation/resampling methods between IM and PS.

**Reply by le0daniel** [reply]
Hey, thx for the Input. I get a RMSE of 1857.38 (0.0283418%), which might not seem like a lot, but still, there is a visual difference in the darker areas. The gradient seems off. No matter if I use the resize or not...

I installed Magick now on OSX too (v7.0.11-1) and libtiff is 4.3.0
I have the same difference between those images, no matter if on linux or mac

**Comment by snibgo** [comment]
Your "-colorspace sRGB" does nothing, because the image is already nominally sRGB.

Your input has an alpha channel, but is entirely opaque. "-alpha off" removes the useless alpha channel, saving disk space.

PNG compression is designed for graphics, not photos. A different format may be smaller.

By default, when IM saves an image to PNG and the image has a nominal sRGB ICC profile, IM will strip the profile and simply create a sRGB PNG chunk. However, sRGB profiles differ, so for best accuracy, removing a profile is a bad idea. We can exclude the sRGB chunk, ensuring the sRGB profile will be written to the output PNG.
```
magick color_small.tif -alpha off -define png:exclude-chunk=sRGB out.png
```
Is that a better result?

**Reply by le0daniel** [reply]
Thx for your input. 

I already found that on an old forum post, forgot to mention that. Still, the images are off, looks darker and the shadows stronger. I tried to specify the color profiles, still there is a visual difference... I don't know if this is a color space issue or something else to be honest.

**Comment by le0daniel** [comment]
I will close this as I expect there is no way to come closer to PS in that case :)
Thanks a lot @snibgo @fmw42 for your input and help. Really appreciated your input

**Reply by fmw42** [reply]
I do not understand. When I did it, I got exactly the same results as indicated by compare returning 0.

**Reply by le0daniel** [reply]
I updated to the latest version and I get a difference on linux of 0.02 which makes the shadow appear darker. Especially when you compare the images side by side or switch between them... 
The gradient is different, the blue color too, the shadows appear darker. If you look at the images I shared, you see a small difference. 
No matter if I try the solutions proposed. Did you compare it to the Photoshop Image?

**Reply by fmw42** [reply]
I compare my PNG to the PNG exported from PS.
